packages/codecs-strings/src/base16.ts

```diff
@@ -49,14 +54,13 @@ export const getBase16Encoder = (): VariableSizeEncoder<string> =>
 
                 const n1 = charCodeToBase16(c1);
                 const n2 = charCodeToBase16(c2);
-                if (n1 === undefined || (Number.isInteger(c2) && n2 === undefined)) {
+                if (n1 === undefined || (n2 === undefined && !Number.isNaN(c2))) {
```

Comment:
I'm pretty sure that `charCodeAt()` returns only an integer or `NaN`. Replaced this with the faster `isNaN()`.